# 2024 Linux for Bioinformatics Workshop - Day 03
## Project organization 

### <a name="0"></a>
| Questions   | Objectives |
| -- | --               |
| How can I organize my file systems for a new bioinformatics project  | Create a file systems for a new bioinformatics project
 | How can I document my work | Explain what types of files should go in your `docs`, `data` and `results` directories |
|  | Use the `history` command and a text editor like `nano` to document your work on your projec

## Getting your project starting
Effective project organization is crucial for sequencing projects, but it's often neglected in the rush to explore new data. Of course, while it is best to get yourself organized before you even begin your analyses, itâ€™s never too late to start, either. 

Genomics projects often involve numerous files spread across multiple folders. Each computational analysis adds more files, creating challenges when rerunning analyses. 

> **Note**:
Strong documentation is essential for doing reproductive research, and fortunately, recording computational experiments is simpler than documenting lab data. Embracing Copy/Paste, using clear file names, and detailing methods will enhance understanding for both you and collaborators.

We will start by creating a directory that we can use for the rest of the workshop. First navigate to your home directory. Then confirm that you are in the correct directory using the pwd command.

On Unix-like operating systems, you can recursively create multiple nested directories at once using the `mkdir`command with the `-p` flag as follows:

`mkdir -p <path>`

```bash
# Go to your home directory

# Create the parent directoy projects in your ~ 

# Go to the projects directory

```
* Create the following directories `sequencing_sc2/docs`, `sequencing_sc2/data`, `sequencing_sc2/results`, you can also use another project name if you like

* Use `ls -R` to verify that you have created these directories. The `-R` option for ls stands for recursive. This option causes `ls` to return the contents of each subdirectory within the directory iteratively.

## Organzing your data
Before data analysis, it is important to keep a copy of your raw data, ensuring it remains unchanged. Even if you're confident about specific data cleaning steps, uncertainties or errors may arise later, necessitating a return to the original data.  

Results generated from analyses should be stored in a designated `results` folder to prevent confusion between results and raw data files during future review or publication preparations.

The `docs` folder serves as a repository for written analyses, notes on analysis methodologies, and documents pertinent to eventual publication.

Similar to wet-lab analyses, sensitive record-keeping is crucial for computational analyses. Fortunately, documenting computational steps is even simpler than benchwork.

The `history` command efficiently captures all commands executed during project file analysis and manipulation. 

```bash
# View the commands that you have used so far during this session
history 

# View the last 10 lines of your history, hint: use tail
```

* Use the redirect `>` to create a file called `workshop_log_XXXX_XX_XX.txt`. Use the four-digit year, two-digit month, and two digit day, e.g. `workshop_log_2023_02_22.txt`
* You may have noticed that your `history` contains the `history` command itself. To remove this redundancy from our log, use the `nano` text editor to fix the file.

```bash
# Verify by printing the content of the file to the terminal

```
